nat/traversal/pinger.go

```diff
@@ -182,6 +184,15 @@ func (p *Pinger) PingProvider(ip string, port int) error {
 	return nil
 }
 
+func (p *Pinger) waitForPreviousStageResult() bool {
+	for {
+		event := p.natEventWaiter.WaitForEvent()
```

Comment:
so what happens when we block here forever? how do we exit this loop?

Reply:
We're waiting here for port-mapping event here, just like before. Once we receive a port mapping event, we return from this function (2 lines below).

The idea is that we start pinger, and it starts working only after port-mapping fails. I don't like this myself - we could start pinger only when nat fails, but I didn't change this yet in this PR.

Reply:
So why not solve it?

```golang
for {
       // not sure what type of the event we're getting here, so that needs changing
	ec := make(chan nat.Event)
	go func() { ec <- p.natEventWaiter.WaitForEvent()} ()
	select {
	case <-p.stop:
		return
	case event := <- ec 
		if event.Stage == p.previousStage {
			return event.Type == SuccessEventType
		}
	}
}
```